src/PlaywrightSharp/BrowserContext.cs

```diff
@@ -306,6 +307,21 @@ public Task UnrouteAsync(Func<string, bool> url, Action<Route, IRequest> handler
         /// <inheritdoc />
         public Task SetExtraHttpHeadersAsync(Dictionary<string, string> headers) => Channel.SetExtraHttpHeadersAsync(headers);
 
+        /// <inheritdoc />
+        public async Task<StorageState> GetStorageStateAsync(string path = null)
+        {
+            var state = await Channel.GetStorageStateAsync().ConfigureAwait(false);
+
+            if (!string.IsNullOrEmpty(path))
+            {
+                File.WriteAllText(
```

Comment:
Do we want to have any sort of checking for this? i.e. that it doesn't already exists, etc.? Otherwise, this might throw an exception.

Reply:
I thought about that. If we don't validate the user gets an exception. If we do validate we will need to throw an exception. 
In both scenarios, the user gets an exception. Does it worth to validate that?

Reply:
Yeah, that's a fair point. Let's leave it as is, then - at least for now. :-)